firehose.go

```diff
@@ -0,0 +1,93 @@
+package hedwig
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/binary"
+	"io"
+)
+
+type Firehose struct {
+	messageValidator *messageValidator
+}
+
+func (f *Firehose) Deserialize(reader io.Reader) ([]Message, error) {
+	// TODO: loop through contents and return messages
+	var messagePayload []byte
+	runWithTransportMessageAttributes := false
+	var messages []Message
+	if f.messageValidator.encoder.IsBinary() {
+		for {
+			// TLV format: 8 bytes for size of message, n bytes for the actual message
+			msgSize := make([]byte, 8)
+			_, err := reader.Read(msgSize)
+			if err == io.EOF {
+				break
+			} else if err != nil {
+				return nil, err
+			}
+			msgLength := binary.LittleEndian.Uint64(msgSize)
+			messagePayload = make([]byte, msgLength)
```

Comment:
could be optimized by re-using `messagePayload` instead of re-allocating it once every loop. Like so:
```
if len(messagePayload) < msgLength {
    messagePayload = append(messagePayload, make([]byte, msgLength - len(messagePayload))...)
}
n, err = reader.Read(messagePayload[:msgLength])
```